sdk/identity/Azure.Identity/src/OnBehalfOfCredential.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Microsoft.Identity.Client;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    ///
```

Comment:
nit: Doc comment.

Reply:
I should mark this as draft, since this is really a work in progress.

Reply:
Gotcha.